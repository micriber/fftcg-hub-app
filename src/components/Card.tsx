import React from 'react';
import {StyleSheet, View, ViewStyle} from 'react-native';

const baseStyles = {
  card: {
    // width: '100%',
    // flexDirection: 'row',
    // flexWrap: 'wrap',
    // justifyContent: 'space-between',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 1,
    borderColor: '#000',
  },
};

const styles = StyleSheet.create({
  dark: {
    ...baseStyles.card,
  },
  light: {
    ...baseStyles.card,
  },
});

type Props = {
  theme: 'light' | 'dark';
  children: React.ReactNode;
  style: ViewStyle;
};

const Card = ({
  children,
  theme = 'light',
  style = StyleSheet.create({}),
}: Props) => {
  const currentStyles = styles[theme];
  return <View style={[currentStyles, style]}>{children}</View>;
};

export default Card;
