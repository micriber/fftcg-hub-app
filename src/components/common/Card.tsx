import React from 'react';
import {ImageStyle, StyleSheet, TextStyle, View, ViewStyle} from 'react-native';

const baseStyles = {
  card: {
    // flex: 1,
    // width: '100%',
    // flexDirection: 'row',
    // flexWrap: 'wrap',
    // justifyContent: 'space-between',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 1,
    borderColor: '#000',
  },
};

const styles = StyleSheet.create({
  dark: {
    ...baseStyles.card,
  },
  light: {
    ...baseStyles.card,
  },
});

type Props = {
  theme?: 'light' | 'dark';
  children: React.ReactNode;
  style?: ViewStyle | TextStyle | ImageStyle;
};

const Card = ({style, children, theme = 'light'}: Props) => {
  const currentStyles = styles[theme];
  return <View style={[currentStyles, style]}>{children}</View>;
};

export default Card;
