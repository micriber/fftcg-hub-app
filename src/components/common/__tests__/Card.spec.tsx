/**
 * @format
 */

import 'react-native';
import React from 'react';
import Card from '../Card';
import {render} from '@testing-library/react-native';
import {Text} from 'react-native';

// More about testing in this repository:
// https://github.com/bamlab/react-native-testing/blob/master/src/pages/Home/__tests__/Home.test.tsx
describe('[Component] Card', () => {
  it('renders correctly', async () => {
    const root = render(
      <Card>
        <Text>Children</Text>
      </Card>,
    );

    expect(root.toJSON()).toMatchSnapshot();
  });
});
