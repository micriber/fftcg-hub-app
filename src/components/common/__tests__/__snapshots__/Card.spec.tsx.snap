// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Card renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#000",
        "elevation": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
      undefined,
    ]
  }
>
  <Text>
    Children
  </Text>
</View>
`;
